code/game/objects/explosion.dm

```diff
@@ -95,6 +95,27 @@
 		for (var/turf/T in trange(max_range, epicenter))
 			var/dist = cheap_pythag(T.x - x0, T.y - y0)
 
+			if(explosion_newmethod)	//realistic explosions that take obstacles into account
+				var/turf/Trajectory = T
+				while(Trajectory != epicenter)
+					Trajectory = get_step_towards(Trajectory,epicenter)
+					if(istype(Trajectory,/turf/simulated/wall) || istype(Trajectory,/turf/simulated/shuttle/wall))

```

Comment:
You can probably fix all this snowflake by giving atoms an explosion_resistance variable, and distance simply goes up by how resistant the atom is (for the case of Poddoors below going up by 2, etc.)